src/runtime/opencl/opencl_device_api.cc

```diff
@@ -123,29 +119,42 @@ void OpenCLWorkspace::CopyDataFromTo(const void* from,
                                      TVMContext ctx_to,
                                      TVMType type_hint,
                                      TVMStreamHandle stream) {
+  cl_event *pevent = nullptr;
+  size_t nr_events = 0;
+
   this->Init();
   CHECK(stream == nullptr);
+
+  if (IsOpenCLDevice(ctx_from)) {
+    // wait on the last kernel event to complete.
+    OpenCLBuffer *buf = static_cast<OpenCLBuffer*>((void*)from);  // NOLINT(*)
+    if (buf->last_kernel_event) {
+      nr_events++;
+      pevent = &buf->last_kernel_event;
```

Comment:
I'm not sure but my understanding is that `event_wait_list`(pevent) should be an array of wait events and  `num_events_in_wait_list`(nr_events) is the number of wait events. `nr_events` is incremented if `last_kernel_event` is set. But does it reflects the number of wait events?

Reply:
Yes, nr_events equals either 0 or 1 here.  nr_events equals 1 only when last_kernel_event is set - we have to wait the kernel execution related to the `from` buffer to be completed

Reply:
Ok, I got it. If nr_events expects either 0 or 1, I'd suggest `nr_events = 1` not increment.